sources/api/schnauzer/src/helpers.rs

```diff
@@ -1346,9 +1360,177 @@ handlebars_helper!(any_enabled: |arg: Value| {
+
+    // Check number of parameters, must be exactly two (OCI spec section to render and settings values for the section)
+    debug!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+    debug!("params: {:?}", helper.params());
+
+    debug!("Getting the requested OCI spec section to render");
+    let oci_defaults_values = get_param(helper, 0)?;
+    // We want the settings path so we know which OCI spec section we are rendering.
+    // e.g. settings.oci-defaults.resource-limits
+    let settings_path = get_param_key_name(helper, 0)?;
+    // Extract the last part of the settings path, which is the OCI spec section we want to render.
+    let oci_spec_section = settings_path
+        .split('.')
+        .last()
+        .expect("did not find (got None for) an oci_spec_section");
+
+    // Render the requested OCI spec section
+    let section =
+        OciSpecSection::from_str(oci_spec_section).context(error::InvalidOciSpecSectionSnafu)?;
+    let result_lines = match section {
+        OciSpecSection::Capabilities => oci_spec_capabilities(oci_defaults_values)?,
+        OciSpecSection::ResourceLimits => oci_spec_resource_limits(oci_defaults_values)?,
+    };
+
+    // Write out the final values to the configuration file
+    out.write(result_lines.as_str())
+        .context(error::TemplateWriteSnafu {
+            template: template_name.to_owned(),
+        })?;
+
+    Ok(())
+}
+
+/// This helper writes out the capabilities section of the default
+/// OCI runtime spec that `containerd` will use.
+///
+/// This function is called by the `oci_defaults` helper function,
+/// specifically when matching over the `oci_spec_section` parameter
+/// to determine which section of the OCI spec to render.
+///
+/// This helper function generates the linux capabilities section of
+/// the OCI runtime spec from the provided `value` parameter, which is
+/// the settings data from the datastore (`settings.oci-defaults.capabilities`).
+fn oci_spec_capabilities(value: &Value) -> Result<String, RenderError> {
+    let oci_default_capabilities: HashMap<OciDefaultsCapability, bool> =
+        serde_json::from_value(value.clone())?;
+
+    // Output the capabilities that are enabled
+    let capabilities_lines: Vec<String> = oci_default_capabilities
+        .iter()
+        .filter(|(_, &capability_enabled)| capability_enabled)
+        .map(|(&capability, _)| format!("\"{}\"", capability.to_linux_string()))
+        .collect();
+
+    let capabilities_lines_joined = capabilities_lines.join(",\n");
+
+    let capabilities_section = format!(
+        "\"bounding\": [
+{capabilities_bounding}
+],
+\"effective\": [
+{capabilities_effective}
+],
+\"permitted\": [
+{capabilities_permitted}
+]",
+        capabilities_bounding = capabilities_lines_joined,
+        capabilities_effective = capabilities_lines_joined,
+        capabilities_permitted = capabilities_lines_joined,
+    );
```

Comment:
You can use a combination of raw strings and `concat!` to clean this up so the string literal isn't flush with the left margin:
```suggestion
    let capabilities_section = format!(
        concat!(
            r#""bounding": ["#,
            "{capabilities_bounding}",
            "]",
            r#""effective": ["#,
            "{capabilities_effective}",
            "]",
            r#""permitted": ["#,
            "{capabilities_permitted}",
            "]",
        ),
        capabilities_bounding = capabilities_lines_joined,
        capabilities_effective = capabilities_lines_joined,
        capabilities_permitted = capabilities_lines_joined,
    );
```

(same for the rlimit function)

Reply:
This is much nicer, thank you!